core/app/models/spree/shipment.rb

```diff
@@ -14,6 +21,8 @@ class Shipment < Spree::Base
     has_many :state_changes, as: :stateful
     has_many :cartons, -> { distinct }, through: :inventory_units
 
+    validates :state, inclusion: { in: [READY, PENDING, SHIPPED, CANCELED] }
```

Comment:
We should extract this into a method, as stores might want to extend/change states.